platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -792,7 +843,17 @@ def get_firmware_update_notification(self, image_path):
         return "Immediate power cycle is required to complete {} firmware update".format(self.name)
 
     def install_firmware(self, image_path):
-        return self.__install_firmware(image_path)
```

Comment:
@alexrallen suggest to define a separate API `install_firmware_no_reboot`

Reply:
Will address your comments regarding a separate API for functions I am calling from the install pending script in the other thread.